practice/intlist.c

```diff
@@ -0,0 +1,172 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// intlist.c
+// Program for Linked List functions.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include "intlist.h"
+// Function to create a new linked list
+LinkedList* Create () {
+	LinkedList* list = (LinkedList*)malloc (sizeof (LinkedList)); // Allocate memory for the list structure
+	if (list == NULL) {
+		printf ("Memory allocation failed\n");
+		return NULL;
+	}
+	list->head = NULL; // Initialize the head pointer to NULL, Indicating an empty list
+	return list;
+}
+
+// Function to add an element to the end of the list
+void Add (LinkedList* list, int val) {
+	Node* newNode = (Node*)malloc (sizeof (Node)); // Allocate memory for the new node
+	if (newNode == NULL) {
+		fprintf (stderr, "Memory allocation failed!\n");
+		exit (EXIT_FAILURE);
+	}
+	newNode->data = val;
+	newNode->next = NULL;
+	if (list->head == NULL) {
+		list->head = newNode;
+	}
+	else {
+		Node* current = list->head;
+		while (current->next != NULL) {
+			current = current->next;
+		}
+		current->next = newNode;
+	}
+}
+
+// Function to insert an element at a particular index (zero-based)

```

Comment:
Comments about what a method does should be added within a ///< summary>...< /summary> tag so you or the person using the method can hover over each method and get to know it's description.
// comments are for explaining code logic (meant for developers editing the method for eg.)

Reply:
Please fix this throughout the code